odp/core/snowflake.py

```diff
@@ -88,8 +89,10 @@ def get_snowflake_queries(
     AND START_TIME < %(before_datetime)s
     AND ERROR_CODE IS NULL
 ORDER BY START_TIME DESC
-LIMIT 10000;
-        """
+"""
+    if limit is not None:
+        sql += f"LIMIT {limit}"
```

Comment:
To be safe I usually add a leading space. I know we have the new line above but still feels safer.